contracts/eosio.system/producer_pay.cpp

```diff
@@ -120,34 +98,34 @@ namespace eosiosystem {
          eosio_assert(now() >= prod->last_rewards_claim + seconds_per_day, "already claimed rewards within a day");
       }
 
-      /// calcualte the price-per-block 
-      
-      /*
-      eosio::asset rewards = prod->per_block_payments;
-      
-      if ( _global.exists() ) {
-         auto parameters = _global.get();
-         if ( parameters.eos_bucket.amount > 0 && prod->total_votes > 0 ) {
-            eosio::asset standby_payment = payment_per_vote( owner, prod->total_votes, parameters.eos_bucket );
-            if ( standby_payment.amount > 0 ) {
-               rewards               += standby_payment;
-               parameters.eos_bucket -= standby_payment;
-               _global.set( parameters, _self );
-            }
-         }
-      }
+      auto parameters = _global.get();      
+      const asset token_supply = token( N(eosio.token)).get_supply(symbol_type(system_token_symbol).name() );
+      const time time_since_last_fill = now() - parameters.last_pervote_bucket_fill;
+
+      const asset to_eos_bucket = supply_growth( standby_rate, token_supply, time_since_last_fill );
+      const asset to_savings    = supply_growth( continuous_rate - (perblock_rate + standby_rate), token_supply, time_since_last_fill );
+      const asset perblock_pay  = payment_per_block( perblock_rate, token_supply, prod->produced_blocks );
+      const asset issue_amount  = to_eos_bucket + to_savings + perblock_pay;
       
-      eosio_assert( rewards > asset(0, S(4,EOS)), "no rewards available to claim" );
+      INLINE_ACTION_SENDER(eosio::token, issue)( N(eosio.token), {{N(eosio),N(active)}},
```

Comment:
the memo is wrong here, some of these tokens are to savings.

Reply:
Changed memo